win32ss/gdi/gdi32/objects/text.c

```diff
@@ -18,18 +25,21 @@ TextOutA(
     ANSI_STRING StringA;
     UNICODE_STRING StringU;
     BOOL bResult;
+    INT cchWide;
 
     if (lpString != NULL)
     {
         RtlInitAnsiString(&StringA, (LPSTR)lpString);
         RtlAnsiStringToUnicodeString(&StringU, &StringA, TRUE);
+        cchWide = MultiByteToWideChar(CP_ACP, 0, lpString, cchString, NULL, 0);
```

Comment:
???? Isn't this length correctly computed by RtlAnsiStringToUnicodeString, when it performs the conversion? (the length is inside StringU.Length )

Reply:
TextOut uses not null-terminated string. So we shouldn't use RtlAnsiStringToUnicodeString function at here. I will try with MultiByteToWideChar.

Reply:
RtlAnsiStringToUnicodeString handles strings without null termination perfectly fine. That is in fact one of the main advantages of ANSI_STRING/UNICODE_STRING.

Reply:
Thanks, @ThFabba. I reused RtlAnsiStringToUnicodeString.